titus-server-gateway/src/main/java/com/netflix/titus/gateway/eviction/EvictionResource.java

```diff
@@ -0,0 +1,140 @@
+        );
+    }
+
+    @ApiOperation("Return the global eviction quota")
+    @Path("quotas/global")
+    @GET
+    public EvictionQuota getGlobalEvictionQuota() {
+        return Responses.fromSingleValueObservable(
+                evictionServiceClient
+                        .getEvictionQuota(Reference.global())
+                        .map(GrpcEvictionModelConverters::toGrpcEvictionQuota)
+        );
+    }
+
+    @ApiOperation("Return a tier eviction quota")
+    @Path("quotas/tiers/{tier}")
+    @GET
+    public EvictionQuota getTierEvictionQuota(@PathParam("tier") String tier) {
+        return Responses.fromSingleValueObservable(
+                evictionServiceClient
+                        .getEvictionQuota(Reference.tier(StringExt.parseEnumIgnoreCase(tier, Tier.class)))
+                        .map(GrpcEvictionModelConverters::toGrpcEvictionQuota)
+        );
+    }
+
+    @ApiOperation("Return a capacity group eviction quota")
+    @Path("quotas/capacityGroups/{name}")
+    @GET
+    public EvictionQuota getCapacityGroupEvictionQuota(@PathParam("name") String capacityGroupName) {
+        return Responses.fromSingleValueObservable(
+                evictionServiceClient
+                        .getEvictionQuota(Reference.capacityGroup(capacityGroupName))
+                        .map(GrpcEvictionModelConverters::toGrpcEvictionQuota)
+        );
+    }
+
+    @ApiOperation("Terminate a task using the eviction service")
+    @Path("tasks/{id}")
+    @DELETE
+    public TaskTerminateResponse terminateTask(@PathParam("id") String taskId,
+                                               @QueryParam("reason") String reason) {
+        return Responses.fromSingleValueObservable(
+                evictionServiceClient
+                        .terminateTask(taskId, reason)
+                        .toObservable()
+                        .materialize()
+                        .flatMap(event -> {
+                            switch (event.getKind()) {
+                                case OnError:
+                                    if (event.getThrowable() instanceof EvictionException) {
+                                        return Observable.just(TaskTerminateResponse.newBuilder()
+                                                .setAllowed(false)
+                                                .setReasonCode("failure")
+                                                .setReasonMessage(event.getThrowable().getMessage())
+                                                .build()
+                                        );
+                                    }
+                                    return Observable.error(event.getThrowable());
+                                case OnCompleted:
+                                    return Observable.just(TaskTerminateResponse.newBuilder()
+                                            .setAllowed(true)
```

Comment:
should there be a reason code for successful?

Reply:
Yes, I will add it.